### PR TITLE
enhance mixed precision

### PR DESCRIPTION
**Pull request by fsx950223**

### CODE REVIEW

File: efficientdet/hparams_config.py [hunk 1]
```diff
@@ -222,8 +222,7 @@ def default_detection_configs():
   h.weight_decay = 4e-5
   # use horovod for multi-gpu training. If None, use TF default.
   h.strategy = None  # 'tpu', 'horovod', None
-  # precision: one of 'float32', 'mixed_float16', 'mixed_bfloat16'.
-  h.precision = None  # If None, use float32.
+  h.mixed_precision = False  # If False, use float32.
```
Comment (mingxingtan) [inline efficientdet/hparams_config.py:225]: Isn't precision is good enough? Do we need this extra flag? (it seems to introduce some extra code complexity)

File: efficientdet/aug/autoaugment.py [hunk 2]
```diff
@@ -38,6 +37,10 @@
 except ImportError:
   from tensorflow.contrib import image as image_ops  # pylint: disable=g-import-not-at-top
 
+try:
+  import tensorflow_probability as tfp
+except ImportError:
+  import tensorflow.distributions as tfp
```
Comment (mingxingtan) [inline efficientdet/aug/autoaugment.py:43]: Could you add a simple comment to these two comments? (which one is recommended by tf2?)

File: efficientdet/hparams_config.py (hunk 1 above)
Comment (fsx950223) [reply efficientdet/hparams_config.py:225]: According to https://github.com/google/automl/issues/488.
TPUs don't support float16?

File: efficientdet/inference.py [hunk 3]
```diff
@@ -162,10 +162,18 @@ def build_model(model_name: Text, inputs: tf.Tensor, **kwargs):
...
+    pp = 'float32'
   cls_outputs, box_outputs = utils.build_model_with_precision(
-      kwargs.get('precision', None), model_arch, inputs, False, model_name,
+      pp, model_arch, inputs, False, model_name,
       **kwargs)
-  if kwargs.get('precision', None):
+  if mixed_precision:
     # Post-processing has multiple places with hard-coded float32.
     # TODO(tanmingxing): Remove them once post-process can adpat to dtypes.
     cls_outputs = {k: tf.cast(v, tf.float32) for k, v in cls_outputs.items()}
```
Comment (fsx950223) [inline efficientdet/inference.py:179]: Could remove them?

**Comment by fsx950223** [comment]
I try to fix the tutorial. But my editor rewrite the 90% tutorial.
I'm afraid that I can't do it anymore.

**Pull request merged**